overlord/hookstate/hooktest/handler.go

```diff
@@ -41,6 +45,10 @@ func NewMockHandler() *MockHandler {
 
 // Before satisfies hookstate.Handler.Before
 func (h *MockHandler) Before() error {
+	executing := atomic.AddInt32(&h.Executing, 1)
+	if executing != 1 {
+		panic(fmt.Sprintf("More than one handler executed: %d", executing))
```

Comment:
This test deliberately reuses single instance of mock handler and relies on its implementation to verify that multiple hooks are not run concurrently and that SetBlocked() lambda does its job. This test will fail every time if SetBlocked implementation is removed.

I agree it would be nice to have a test where there is a second snap, and its hook can be run at the same time and independent of first snap' hook. But I'm not sure how to make such a test right now without introducing flakiness.